y3prediction/prediction.py

```diff
@@ -1247,9 +1254,13 @@ def compute_smoothed_char_length(href_fluid, href_wall):
     compute_smoothed_char_length_compiled = \
         actx.compile(compute_smoothed_char_length)
 
-    smoothed_char_length, smoothed_char_length_wall = \
-        compute_smoothed_char_length_compiled(char_length_nodes,
-                                              char_length_wall_nodes)
+    use_smoothed_charlen = False
+    if use_smoothed_charlen:
+        smoothed_char_length, smoothed_char_length_wall = \
+            compute_smoothed_char_length_compiled(char_length_nodes,
+                                                  char_length_wall_nodes)
+        char_len_fluid_nodes = force_evaluation(actx, smoothed_char_length)
+        char_len_wall_nodes = force_evaluation(actx, smoothed_char_length_wall)
```

Comment:
FWIW, I think all of the `force_evaluation`s here are redundant except for the `char_length_*` ones. (`mask_from_elements` does everything on the host except for a `from_numpy` at the end, `thaw` doesn't do any computation, and the result from a compiled function should already be evaluated AFAIK.)

Reply:
I don't doubt it.  There is a high penalty paid for missing one, so I superstitiously hit everything that even remotely seemed questionable.  We can nail it down before making the actual changes.   The main thing I was trying to do here is make sure everything that is constant was made of numbers, with no graph or expression, even a simple one.